var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -46,9 +46,11 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
         'serial': [True,  'Whether to build serial backend'],
         'rocm': [False, 'Whether to build HIP backend'],
         'sycl': [False, 'Whether to build the SYCL backend'],
+        'openmptarget': [False, 'Whether to build the OpenMPTarget backend']
     }
     conflicts("+rocm", when="@:3.0")
     conflicts("+sycl", when="@:3.3")
+    conflicts("+openmptarget", when="@:3.5")
```

Comment:
Around line 196 in this file we have some logic to check that the correct std flag is passed for recent cuda versions, I think we should have something similar for rocm, sycl and openmptarget. This would allow the user to get a quick error at concretization instead of waiting for cmake to fail

Reply:
I am not aware of any restriction for `HIP`. I added the checks for `SYCL` and `OpenMPTaregt`, though, to match what we do in `CMake`.